src/main/java/org/javarosa/core/model/TriggerableDag.java

```diff
@@ -645,6 +644,28 @@ void deleteRepeatInstance(FormInstance mainInstance, EvaluationContext evalConte
         return result;
     }
 
+    /**
+     * Builds an index to look up relevance for repeats. Assumes that there can only be a single Triggerable associated
+     * with a repeat and that it must represent relevance. This Triggerable is evaluated directly to determine relevance
+     * at a new repeat prompt. This is used instead of the relevance stored in the corresponding TreeElement and updated
+     * by the DAG because repeat templates aren't updated by DAG recomputations. Additionally, the context intersection
+     * made in Triggerable.intersectContextWith means we may not evaluate against the correct context when the triggerable is
+     * applied.
+     */
+    private static Map<TreeReference, QuickTriggerable> buildRelevancePerRepeat(FormInstance mainInstance, Set<QuickTriggerable> triggerables) {
+        Map<TreeReference, QuickTriggerable> relevancePerRepeat = new HashMap<>();
+        for (QuickTriggerable triggerable : triggerables)
+            if (triggerable.isCondition())
+                for (TreeReference target : triggerable.getTargets())
+                    if (mainInstance.getTemplate(target) != null)
```

Comment:
If it has a template it's a repeat right? I think we need to express that in a clearer way throughout.